index.ts

```diff
@@ -45,5 +45,22 @@ export function patternToRegex(...matchPatterns: readonly string[]): RegExp {
 		return allStarsRegex;
 	}
 
-	return new RegExp(matchPatterns.map(x => getRawRegex(x)).join('|'));
+	return new RegExp(matchPatterns.map(x => getRawPatternRegex(x)).join('|'));
+}
+
+function getRawGlobRegex(globPattern: string): string {
+	return globPattern
+		.replace(/[|\\{}()[\]^$+*?.]/g, '\\$&') // Escape
+		.replace(/-/g, '\\x2d') // Escape
```

Comment:
This concerns me because the other function doesn't have it. Perhaps it should just import: https://github.com/sindresorhus/escape-string-regexp

I'm fine with the extra dependency